### PR TITLE
Add helper to convert collective page sections to new format

### PR DESCRIPTION
**Pull request by Betree**
Related to https://github.com/opencollective/opencollective/issues/3744

### CODE REVIEW

File: lib/collective-sections.js [hunk 1]
```diff
@@ -319,3 +319,77 @@ export const getFilteredSectionsForCollective = (collective, isAdmin, isHostAdmi
...
+  ABOUT: 'ABOUT',
+  BUDGET: 'BUDGET',
+  CONNECT: 'CONNECT',
+  CONTRIBUTE: 'CONTRIBUTE',
+  OTHER_INITIATIVES: 'OTHER_INITIATIVES', // Events, projects, connected collectives
+};
+
+/**
+ * Map sections to their categories. Any section that's not in this object will be considered
+ * as a "Widget" (aka. a section without navbar category).
```
Comment (sbinlondon) [inline lib/collective-sections.js:333]: 🙌  Great idea

File: lib/collective-sections.js [hunk 2]
```diff
@@ -319,3 +319,77 @@ export const getFilteredSectionsForCollective = (collective, isAdmin, isHostAdmi
...
+  type: 'SECTION',
+  name: section,
+  isEnabled,
+  restrictedTo,
+});
+
+/**
+ * Converts legacy sections to their new format
+ */
+export const convertSectionsToNewFormat = sections => {
```
Comment (sbinlondon) [inline lib/collective-sections.js:368]: And where will we use this? In `getFilteredSectionsForCollective`?

**Review by sbinlondon — APPROVED** [review]
Looks great, thanks for this!

File: lib/collective-sections.js (hunk 2 above)
Comment (Betree) [reply lib/collective-sections.js:368]: Yes, the places were we use `getFilteredSectionsForCollective` should be able to handle both sections format while we transition

**Pull request merged**